nautilus_trader/model/orders/stop_market.pyx

```diff
@@ -337,12 +337,12 @@ cdef class StopMarketOrder(Order):
             client_order_id=init.client_order_id,
             order_side=init.side,
             quantity=init.quantity,
-            trigger_price=Price.from_str_c(init.options["trigger_price"]),
+            trigger_price=Price.from_str_c(init.options['trigger_price']),
             trigger_type=trigger_type_from_str(init.options["trigger_type"]),
-            time_in_force=init.time_in_force,
-            expire_time_ns=init.options["expire_time_ns"],
             init_id=init.id,
             ts_init=init.ts_init,
+            time_in_force=init.time_in_force,
+            expire_time_ns=init.options["expire_time_ns"] if init.options["expire_time_ns"] is not None else 0,
```

Comment:
- Double indexing
- I think `expire_time_ns` should be an integer?
- Do we really need to reorder the params at this stage?

Reply:
As you can see there nothing much changed order that `None` check was added, because it could happen that `init.options["expire_time_ns"] ` is None

Reply:
Basically this 
```
expire_time_ns=init.options["expire_time_ns"],
```

became this 
```
expire_time_ns=init.options["expire_time_ns"] if init.options["expire_time_ns"] is not None else 0,
```

Reply:
Personally wouldnt worry much about this, as you once said this was bad design to include some of this params into `option` dictionary.  But we have some trouble unifying it with Rust as there every param in `options` is its own field

Reply:
I don't see anywhere it can be `None` as its typed as `uint64_t expire_time_ns`. But doesn't matter so much I can check this once merged.

Reply:
yeah, I think the `options` dict itself wasn't a good design choice.

Reply:
`None` can come from Rust I think, and i don't see much problem fixing it here